src/tree/test_treemath.rs

```diff
@@ -19,29 +24,33 @@ fn verify_binary_test_vector_treemath() {
     let parent: Vec<u32> = decode_vec(VecSize::VecU32, cursor).unwrap();
     let sibling: Vec<u32> = decode_vec(VecSize::VecU32, cursor).unwrap();
 
+    fn test_result(index: usize, input: &u32, result: Result<NodeIndex, TreeMathError>) {
+        if index != *input as usize {
+            assert!(result.is_ok());
+            assert_eq!(NodeIndex::from(*input), result.unwrap());
+        } else {
+            assert!(result.is_err());
+        }
+    }
+
     for (i, r) in root.iter().enumerate() {
         assert_eq!(NodeIndex::from(*r), treemath::root(LeafIndex::from(i + 1)));
     }
     for (i, l) in left.iter().enumerate() {
-        assert_eq!(NodeIndex::from(*l), treemath::left(NodeIndex::from(i)));
+        let result = treemath::left(NodeIndex::from(i));
```

Comment:
Can you add a comment to each of these test cases to make clear what they are supposed to test?